src/app/modules/submission/email-submission/email-submission.util.ts

```diff
@@ -535,18 +541,172 @@ export const maskUidOnLastField = (
         autoReplyField.question === SPCPFieldTitle.CpUid && // Check field title
         index === autoReplyData.length - 1 // Check field position
       ) {
-        return {
-          question: autoReplyField.question,
-          answerTemplate: autoReplyField.answerTemplate.map((answer) => {
-            return answer.length >= 4 // defensive, in case UID length is less than 4
-              ? '*'.repeat(answer.length - 4) + answer.substr(-4)
-              : answer
-          }),
-        }
+        return maskField(autoReplyField)
       } else {
         return autoReplyField
       }
     },
   )
   return maskedAutoReplyData
 }
+
+/**
+ * Function to generate email data
+ * for a single field
+ */
+const createFormattedDataForOneField = <T>(
+  response: ProcessedFieldResponse,
+  hashedFields: Set<string>,
+  getFormattedFunction: (
+    response: ResponseFormattedForEmail,
+    hashedFields: Set<string>,
+  ) => T,
+): T[] => {
+  if (isProcessedTableResponse(response)) {
+    return getAnswerRowsForTable(response).map((row) =>
+      getFormattedFunction(row, hashedFields),
+    )
+  } else if (isProcessedCheckboxResponse(response)) {
+    const checkbox = getAnswerForCheckbox(response)
+    return [getFormattedFunction(checkbox, hashedFields)]
+  } else {
+    return [getFormattedFunction(response, hashedFields)]
+  }
+}
+
+/**
+ * Helper function to mask NRICs in Corppass Validated UID
+ */
+const maskField = (field: EmailAutoReplyField): EmailAutoReplyField => {
```

Comment:
I wonder if it might be better to make this function signature more generic, perhaps taking a string and a length with a default value of 4, and having it return the masked string

Reply:
edited